controllers/helmchart_controller.go

```diff
@@ -508,22 +538,53 @@ func (r *HelmChartReconciler) reconcileFromTarballArtifact(ctx context.Context,
 	// or write the chart directly to storage.
 	pkgPath := chartPath
 	isValuesFileOverriden := false
-	if chart.Spec.ValuesFile != "" {
-		srcPath, err := securejoin.SecureJoin(tmpDir, chart.Spec.ValuesFile)
-		if err != nil {
-			return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
+	if chart.Spec.ValuesFile != "" || len(chart.Spec.ValuesFiles) > 0 {
+		var valuesFiles []string
+		var valuesMap map[string]interface{}
+
+		// Prepend a list of valuesFiles with the deprecated ValuesFile.
+		if chart.Spec.ValuesFile != "" {
+			valuesFiles = append(valuesFiles, chart.Spec.ValuesFile)
 		}
-		if f, err := os.Stat(srcPath); os.IsNotExist(err) || !f.Mode().IsRegular() {
-			err = fmt.Errorf("invalid values file path: %s", chart.Spec.ValuesFile)
-			return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
+		valuesFiles = append(valuesFiles, chart.Spec.ValuesFiles...)
+
+		for _, v := range valuesFiles {
+			srcPath, err := securejoin.SecureJoin(tmpDir, v)
+			if err != nil {
+				return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
+			}
+			if f, err := os.Stat(srcPath); os.IsNotExist(err) || !f.Mode().IsRegular() {
+				err = fmt.Errorf("invalid values file path: %s", v)
+				return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
+			}
+
+			valuesData, err := ioutil.ReadFile(srcPath)
+			if err != nil {
+				err = fmt.Errorf("failed to read from values file '%s': %w", v, err)
+				return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
+			}
```

Comment:
```suggestion
			var valuesData []byte
			cfn := filepath.Clean(v)
			for _, f := range helmChart.Files {
				if f.Name == cfn {
					valuesData = f.Data
					break
				}
			}
			if valuesData == nil {
				err = fmt.Errorf("invalid values file path: %s", v)
				return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
			}
```
Replacing the file system loading with this causes the current integration tests to fail @nniikkoollaaii

Reply:
OK  then this is the cause the original implementations used different approaches too. 

Sry i wasn't able to see this :/

Probably you can reuse some of my additional checks in the unit tests like [here](https://github.com/nniikkoollaaii/source-controller/blob/values-files/controllers/helmchart_controller_test.go#L809)?

Reply:
Ya I didn't look into why or how it fails. I will add the tests! 😁 I think we can clean up the funcs after.

Reply:
I think it's because the example files weren't updated for the new paths which makes sense. I don't think we would be able to fix this now, because it'll break current users of `ValuesFile`.